uPortal-url/src/main/java/org/apereo/portal/url/AbstractUrlBuilder.java

```diff
@@ -27,17 +26,7 @@
     private final Map<String, String[]> parameters;
 
     public AbstractUrlBuilder() {
-        this.parameters =
-                MapConstraints.constrainedMap(
-                        new ParameterMap(),
-                        new MapConstraint<String, String[]>() {
-                            @Override
-                            public void checkKeyValue(String key, String[] value) {
-                                Validate.notNull(key, "name can not be null");
-                                Validate.noNullElements(
-                                        value, "values can not be null or contain null elements");
-                            }
-                        });
+        this.parameters = new HashMap<>();
     }
 
     @Override
```

Comment:
One new exposure is that `getParameters()` is returning the unprotected map that no longer has the checks. Wrap it in an unmodifiableMap.

Reply:
Apply the same treatment in PortletUrlBuilder.getPublicRenderParameters()

Reply:
Updated